blog/_posts/2016-10-06-julia-0.5-highlights.md

```diff
@@ -0,0 +1,726 @@
+
+    julia> T[2,:,2] # vector
+    4-element Array{Int64,1}:
+     11
+     14
+     19
+     26
+
+The only example from above that doesn't produce a vector is the last one:
+
+    julia> T[:,4,:]
+    3×2 Array{Int64,2}:
+     18  25
+     19  26
+     20  27
+
+The result is a matrix since the leading and trailing slices are ranges, and the middle slices disappears since it is scalar, leaving a matrix.
+In 0.5, slicing behavior is also naturally generalized to higher dimensional slices, beyond 0-dimensional (scalar) and 1-dimensional indices:
+
+    julia> I = [1 2 1; 1 3 2]
+
+    julia> J = [4 2 1 3]
+    1×4 Array{Int64,2}:
+     4  2  1  3
+
+    julia> M[I,J]
+    2×3×1×4 Array{Int64,4}:
+    [:, :, 1, 1] =
+     17  18  17
+     17  19  18
+
+    [:, :, 1, 2] =
+     5  6  5
+     5  7  6
+
+    [:, :, 1, 3] =
+     2  3  2
+     2  4  3
+
+    [:, :, 1, 4] =
+     10  11  10
+     10  12  11
+
+We have the following natural identity on dimensions:
+
+    size(M[I,J]) == (size(I,1), size(I,2), size(J,1), size(J,2))
+
+In addition to being more systematic and symmetrical, this new behavior allows many complex indexing operations to be expressed concisely.
+
+Although the change to multidimensional slicing behavior is a significant breaking change, it has caused surprisingly little havoc in the Julia package ecosystem.
+It tends to primarily affect linear algebra code, and when code does break, it's usually fairly clear what is broken and what needs to be done to fix it.
+Unfortunately, there's no way to make this change without breaking some code – we apologize in advance for the inconvenience, and we hope you find the improvement to be worthwhile worthwhile.
+
+### Array views
+
+One of the major news items of 0.5 is a non-change:
+array slices still create copies of array data.
+There was a lot of discussion about changing the default behavior to creating views, but we ended up deciding against this change and keeping the old behavior.
+The motivation for views by default was to improve performance drastically in a variety of slow cases, but after a lot of discussion, experiments, and benchmarks, it was decided not to make this change.
+The conversation about this decision is long, so I'll summarize the major points:
+
+- Slicing should either consistently produce views or copies.
+Unpredictably doing one or the other depending on the type – or worse still, on runtime values – would be a disaster for writing reliable generic code.
+
+- Guaranteeing view semantics for all abstract arrays – especially sparse and custom array types – is hard and can be quite slow and/or expensive in general cases.
+
+- Even in the case of dense arrays with cheap array views, it's not clear that views are always a performance win.
+In some cases they definitely are, but in others the fact that a copied slice is contiguous and has optimal memory ordering for iteration overwhelms the benefit of not copying.
+Benchmarks were a wash.
```

Comment:
I would drop this sentence since the previous sentence already expresses this idea.